packages/react-instantsearch-nextjs/src/InstantSearchNext.tsx

```diff
@@ -55,6 +69,10 @@ export function InstantSearchNext<
     };
   }, []);
 
+  useEffect(() => {
+    window[InstantSearchLastPath] = pathname;
```

Comment:
won't this also be set if someone has InstantSearch set up with routing for multiple different path names?

Reply:
Not sure I get it, are you refering to a situation where there's an InstantSearch implementation in multiple paths and the user navigates from one to the other?

In that case, this is okay as search needs to be triggered.

In any case I'm not fond of having to store this data globally, but I haven't found a good way to track route changes since `useRouter` events are not available in app routing. I'm definitely open to ideas around that.

Reply:
I'm thinking of a use case where with routing someone has for example `/search/:category` set up as the path. If you refine the category, won't it also be true here and force an extra search?

Reply:
This change caters to that use case specifically. You can see it in this comment https://github.com/algolia/instantsearch/pull/6452/files#r1863228512.

Reply:
It would be nice if we had an InstantSearch singleton for React. It would make sure they only use one instance of InstantSearch and we could probably keep track of URLs in there.